### DISCUSSION TITLE
How can I let cells auto grow in height when merged?

### DISCUSSION THREAD

**Discussion by Gricardov** [exceljs/exceljs#2483] [Q&A]
Hi dear team;
I'm having an issue related to merged cells. When I don't merge them, they grow according to the content; but if I merge them, they always keep the same height. This is basically my code:

setCellValue(value)
worksheet.mergeCells(start+ ":" + end);
worksheet.getCell(start).alignment = { vertical: 'middle', horizontal: 'left',wrapText: true };

if I comment the second line, the cell grow as expected; although they are not merged.

Please help

**Comment by banana618859** [comment]
I try to anthor way to resolve this problem.
just set row height，i will compute the word length and row count
`worksheet.getRow(i).height = word length / row contain length * 13.5`

**Comment by DinoZenka** [comment]
> I try to anthor way to resolve this problem.
> just set row height，i will compute the word length and row count
> `worksheet.getRow(i).height = word length / row contain length * 13.5`

Hello, how did you compute the word length?

**Comment by gsmirnov94** [comment]
Hi! My solution is:
```
_reCalcRowHeights = () => {
        const koef = 1.4;
        const worksheet = this._worksheet;
        for (let i = 0; i < worksheet._rows.length; i++) {
            const row = worksheet._rows[i];
            let linesCount = 0;
            for (let j = 0; j < row._cells.length; j++) {
                const cell = row._cells[j];
                const cellValue = cell.value || '';
                const columnWidth = worksheet.getColumn(j + 1).width;
                const cellMerges = worksheet._merges[cell.address]?.model;
                if (cellMerges) {
                    if (cellMerges.top !== cellMerges.bottom) {
                        continue;
                    }
                    if (cellMerges.left === cellMerges.right) {
                        linesCount = Math.max(linesCount, Math.ceil(cellValue.length * koef / columnWidth));
                    }
                    if (cellMerges.left < cellMerges.right) {
                        let mergedCellsWidth = 0;
                        for (let k = cellMerges.left; k <= cellMerges.right; k++)
                            mergedCellsWidth += worksheet.getColumn(k).width;

                        linesCount = Math.max(linesCount, Math.ceil(cellValue.length * koef / mergedCellsWidth));
                    }
                }
            }
            row.height = worksheet.properties.defaultRowHeight * linesCount;
        }
    }
```
It runs through the whole worksheet, calculates cells' width (including horizontally merged cells in one row) and calculates row's height.
**Notes**:
1. It does not works with multiline cell velues yet.
2. It does not work with vertically merged cells yet.
3. Change ```koef``` value to tune text lines counter calculation.
